Source/MQTTnet.Extensions.ManagedClient/ManagedMqttClient.cs

```diff
@@ -193,8 +211,31 @@ public Task UnsubscribeAsync(IEnumerable<string> topics)
 
         public void Dispose()
         {
-            _connectionCancellationToken?.Dispose();
-            _publishingCancellationToken?.Dispose();
+            if (_disposed)
+            {
+                return;
+            }
+
+            _disposed = true;
+
+            StopPublishing();
+            StopMaintainingConnection();
+
+            if (_maintainConnectionTask != null)
+            {
+                Task.WaitAny(_maintainConnectionTask);
+                _maintainConnectionTask = null;
```

Comment:
Why is here a "WhenAny" if there is only one? Is there a task missing in the parameter list?

Reply:
Dispose is not async. Therefor I can’t use Task.WhenAny to ensure a task was finished. The synchronous alternative is Task.WaitAny